src/main/java/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -915,6 +965,27 @@ private enum State {
     // The current state.
     private State state = State.READING_VALUE;
 
+    /**
+     * Determines whether the bytes between [start, end) in 'buffer' match the target bytes.
+     * @param target the target bytes.
+     * @param buffer the bytes to match.
+     * @param start index of the first byte to match.
+     * @param end index of the first byte after the last byte to match.
+     * @return true if the bytes match; otherwise, false.
+     */
+    private static boolean bytesMatch(byte[] target, byte[] buffer, int start, int end) {
+        int length = end - start;
+        if (length != target.length) {
+            return false;
+        }
+        for (int i = 0; i < target.length; i++) {
+            if (target[i] != buffer[start + i]) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Suggested TODO—see if it's faster to copy the bytes into a pre-allocated buffer and then perform a comparison. It's possible that a combination of [`System.arraycopy()`](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) and [`Arrays.equals(byte[], byte[])`](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#equals-byte:A-byte:A-) is faster because it can be more easily optimized with native code by the JVM—both are annotated with `@HotSpotIntrinsicCandidate`.

Reply:
Added a TODO. Note: I think it's unlikely for this to be performance critical, since it's only required when a symbol table declaration contains inline symbols. Though theoretically possible, our writers are never actually required to encode symbol tables like this. Also, when inline symbols are enabled via the existing `ALWAYS_INLINE` enum, no symbol table will ever be required.

Reply:
I agree that it's unlikely to be performance critical in this case, but we might find that it makes an impact if we're using inline symbols for writing macro definitions using the template definition language, and we could still need an encoding directive for a macro table even if we're using inline symbols.